packages/config/src/log/messages.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const { WILDCARD_ALL: FUNCTIONS_CONFIG_WILDCARD_ALL } = require('../functions_config')
+
+const { log } = require('./logger')
+
+const warnLegacyFunctionsDirectory = ({ config = {}, logs }) => {
```

Comment:
What are your thoughts on running this after `normalizeConfig()`? This would remove the need for some of the default value assignments and `|| {}`.